## **Soal 1**

**Analisis Harga Rumah dan rata rata luas tanah Berdasarkan Tahun.**

Buatlah program yang menghitung total nilai penjualan (SalePrice) dan rata-rata luas tanah (LotArea) per tahun (YrSold)

## **Soal 2**

**Distribusi Harga Penjualan per Bulan**

Terdapat kolom yang menunjukkan bulan penjualan properti (MoSold) dan tahun penjualan (YrSold). Anda diminta untuk membuat sebuah pivot table yang menampilkan rata-rata harga penjualan properti (SalePrice) berdasarkan bulan penjualan dan tahun penjualan. Tampilkan bagaimana harga rata-rata per bulan berubah dari tahun ke tahun.

## **Soal 3**

**Identifikasi Properti dengan Luas Tanah di Atas Persentil ke-90**

Lakukan Filtering Rumah dengan luas tanah (LotArea) yang lebih besar dari persentil ke-90. Setelah itu, urutkan properti tersebut berdasarkan harga penjualan (SalePrice) secara menurun. Tampilkan properti-properti dengan luas tanah yang paling besar, dan analisis apakah harga penjualan cenderung mengikuti ukuran tanah.

## **Soal 4**

**Analisis Zona Penjualan**

Anda ingin menganalisis jumlah total penjualan properti di setiap zona (MSZoning) berdasarkan tahun penjualan (YrSold). Gunakan tabel silang (crosstab) untuk menghitung jumlah penjualan di setiap zona untuk tiap tahun. Setelah itu, tambahkan kolom baru yang menandai zona dengan penjualan tinggi, di mana zona tersebut memiliki total penjualan di atas rata-rata.

## **Soal 5**

**Analisis Harga per Meter Persegi**

Anda diminta untuk melakukan feature engineering untuk membuat kolom baru bernama price_per_sqft, yang menghitung harga per meter persegi (dari SalePrice dibagi dengan LotArea). Setelah itu, urutkan hasilnya berdasarkan price_per_sqft secara menurun untuk menemukan properti dengan harga per meter persegi tertinggi. Analisis hasil tersebut dan identifikasi properti mana yang memiliki nilai tanah tertinggi.



## Import Modules

In [ ]:
! pip install matplotlib
! pip install seaborn

In [ ]:
# dataframe libraries
import pandas as pd

# array libraries
import numpy as np

# Data Visualization libraries
import matplotlib.pyplot as plt
import seaborn as sns

# Math libraries
import math

## Dataset

In [ ]:
#Membaca dataset untuk Google Colab
df = pd.read_csv('data_cleaning.csv')
df

Id  MSSubClass MSZoning  LotArea Street LotShape LandContour  \
0        1          60       RL     8450   Pave      Reg         Lvl   
1        2          20       RL     9600   Pave      Reg         Lvl   
2        3          60       RL    11250   Pave      IR1         Lvl   
3        4          70       RL     9550   Pave      IR1         Lvl   
4        5          60       RL    14260   Pave      IR1         Lvl   
...    ...         ...      ...      ...    ...      ...         ...   
1333  1456          60       RL     7917   Pave      Reg         Lvl   
1334  1457          20       RL    13175   Pave      Reg         Lvl   
1335  1458          70       RL     9042   Pave      Reg         Lvl   
1336  1459          20       RL     9717   Pave      Reg         Lvl   
1337  1460          20       RL     9937   Pave      Reg         Lvl   

     Utilities LotConfig LandSlope  ... EnclosedPorch 3SsnPorch ScreenPorch  \
0       AllPub    Inside       Gtl  ...             0         0           0   
1       AllPub       FR2       Gtl  ...             0         0           0   
2       AllPub    Inside       Gtl  ...             0         0           0   
3       AllPub    Corner       Gtl  ...           272         0           0   
4       AllPub       FR2       Gtl  ...             0         0           0   
...        ...       ...       ...  ...           ...       ...         ...   
1333    AllPub    Inside       Gtl  ...             0         0           0   
1334    AllPub    Inside       Gtl  ...             0         0           0   
1335    AllPub    Inside       Gtl  ...             0         0           0   
1336    AllPub    Inside       Gtl  ...           112         0           0   
1337    AllPub    Inside       Gtl  ...             0         0           0   

     PoolArea MiscVal  MoSold  YrSold  SaleType  SaleCondition SalePrice  
0           0       0       2    2008        WD         Normal    208500  
1           0       0       5    2007        WD         Normal    181500  
2           0       0       9    2008        WD         Normal    223500  
3           0       0       2    2006        WD        Abnorml    140000  
4           0       0      12    2008        WD         Normal    250000  
...       ...     ...     ...     ...       ...            ...       ...  
1333        0       0       8    2007        WD         Normal    175000  
1334        0       0       2    2010        WD         Normal    210000  
1335        0    2500       5    2010        WD         Normal    266500  
1336        0       0       4    2010        WD         Normal    142125  
1337        0       0       6    2008        WD         Normal    147500  

[1338 rows x 71 columns]

In [ ]:
#Melihat deskripsi singkat dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 71 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             1338 non-null   int64  
 1   MSSubClass     1338 non-null   int64  
 2   MSZoning       1338 non-null   object 
 3   LotArea        1338 non-null   int64  
 4   Street         1338 non-null   object 
 5   LotShape       1338 non-null   object 
 6   LandContour    1338 non-null   object 
 7   Utilities      1338 non-null   object 
 8   LotConfig      1338 non-null   object 
 9   LandSlope      1338 non-null   object 
 10  Neighborhood   1338 non-null   object 
 11  Condition1     1338 non-null   object 
 12  Condition2     1338 non-null   object 
 13  BldgType       1338 non-null   object 
 14  HouseStyle     1338 non-null   object 
 15  OverallQual    1338 non-null   int64  
 16  OverallCond    1338 non-null   int64  
 17  YearBuilt      1338 non-null   int64  
 18  YearRemo

## **Soal 1**

**Analisis Harga Rumah dan rata rata luas tanah Berdasarkan Tahun.**

Buatlah program yang menghitung total nilai penjualan (SalePrice) dan rata-rata luas tanah (LotArea) per tahun (YrSold)


In [ ]:
#Menampilkan 5 baris pertama untuk melihat seluruh dataset secara sekilas
df.head()

Id  MSSubClass MSZoning  LotArea Street LotShape LandContour Utilities  \
0   1          60       RL     8450   Pave      Reg         Lvl    AllPub   
1   2          20       RL     9600   Pave      Reg         Lvl    AllPub   
2   3          60       RL    11250   Pave      IR1         Lvl    AllPub   
3   4          70       RL     9550   Pave      IR1         Lvl    AllPub   
4   5          60       RL    14260   Pave      IR1         Lvl    AllPub   

  LotConfig LandSlope  ... EnclosedPorch 3SsnPorch ScreenPorch PoolArea  \
0    Inside       Gtl  ...             0         0           0        0   
1       FR2       Gtl  ...             0         0           0        0   
2    Inside       Gtl  ...             0         0           0        0   
3    Corner       Gtl  ...           272         0           0        0   
4       FR2       Gtl  ...             0         0           0        0   

  MiscVal  MoSold  YrSold  SaleType  SaleCondition SalePrice  
0       0       2    2008        WD         Normal    208500  
1       0       5    2007        WD         Normal    181500  
2       0       9    2008        WD         Normal    223500  
3       0       2    2006        WD        Abnorml    140000  
4       0      12    2008        WD         Normal    250000  

[5 rows x 71 columns]

In [ ]:
#Jawaban Soal 1 menggunakan groupby
df.groupby(["YrSold"]).agg({
    "SalePrice": "sum",
    "LotArea": "mean"
})

SalePrice       LotArea
YrSold                         
2006     54546130  10651.841924
2007     58336353  11008.019608
2008     50395063  10855.366300
2009     57220176  10468.987013
2010     29389543  10430.743750

In [ ]:
#Jawaban Soal 1 diurutkan berdasarkan nilai kolom SalePrice dari tertinggi ke terendah
df.groupby(["YrSold"]).agg({
    "SalePrice": "sum",
    "LotArea": "mean"
}).sort_values(by = "SalePrice", ascending = False)

SalePrice       LotArea
YrSold                         
2007     58336353  11008.019608
2009     57220176  10468.987013
2006     54546130  10651.841924
2008     50395063  10855.366300
2010     29389543  10430.743750

**Tabel SalePrice dan LotArea untuk tahun yang berbeda memberikan beberapa informasi, yakni sebagai berikut:**


1.   **Hasil Analisis Tren SalePrice**


*   Harga jual turun secara signifikan dari tahun 2007 hingga 2010 dimana harga jual turun dari 58.336.353 pada tahun 2007 menjadi 29.389.543 pada tahun 2010, walaupun luas lot tetap konsisten dari tahun ke tahun.Hal ini mengindikasikan penurunan pasar dalam negeri atau resesi ekonomi.


2.   **Hasil Analisis Tren LotArea**

* Nilai LotArea berkisar antara 10,430.74 hingga 11,008.02 kaki persegi,dan walaupun sedikit berfluktuasi, tetapi tetap stabil. Hal ini berarti ukuran lot yang dijual tidak banyak berubah seiring berjalannya waktu, yang menunjukkan bahwa Harga jual tidak turun karena ukuran luas lot.

3. **Faktor Pasar Potensial**

* Faktor pasar seperti permintaan, kondisi ekonomi, dan preferensi pembeli mungkin menyebabkan SalePrice menurun drastis walaupun luas lot tetap konsisten. Pola ini dapat dijelaskan dengan mempelajari peristiwa eksternal seperti krisis keuangan tahun 2008.

4. **Peluang untuk berinvestasi**

* Terdapat peluang untung berinvestasi pada titik harga yang lebih rendah pada penurunan SalePrice pada tahun 2010. Investor dan pembeli rumah yang mencari nilai real estat dapat memperoleh manfaat dari hal ini.


## **Soal 2**

**Distribusi Harga Penjualan per Bulan**

Terdapat kolom yang menunjukkan bulan penjualan properti (MoSold) dan tahun penjualan (YrSold). Anda diminta untuk membuat sebuah pivot table yang menampilkan rata-rata harga penjualan properti (SalePrice) berdasarkan bulan penjualan dan tahun penjualan. Tampilkan bagaimana harga rata-rata per bulan berubah dari tahun ke tahun.



In [ ]:
#Jawaban nomor 2 mwnggunakan pivot_table dengan bulan menjadi kolom dan tahun menjadi baris
pd.pivot_table(
df,
index=["YrSold"],
columns=["MoSold"],
values="SalePrice",
aggfunc="mean")

MoSold             1              2              3              4   \
YrSold                                                               
2006    201090.000000  194550.000000  190502.291667  184247.750000   
2007    195229.166667  176301.750000  179507.500000  173844.090909   
2008    196187.363636  159370.000000  183379.666667  166244.652174   
2009    213282.600000  187450.000000  181712.500000  185028.000000   
2010    167730.666667  178989.285714  203181.285714  181439.909091   

MoSold             5              6              7              8   \
YrSold                                                               
2006    162907.857143  179218.181818  189463.564516  206477.095238   
2007    173342.951220  182400.403509  200661.229167  190515.083333   
2008    193787.058824  174276.625000  186240.418605  199285.071429   
2009    171852.029412  193688.627451  203379.807018  167971.444444   
2010    186087.627907  184072.142857  135100.000000            NaN   

MoSold             9              10             11             12  
YrSold                                                              
2006    206538.071429  178139.136364  213285.000000  190140.000000  
2007    205737.200000  220190.733333  197614.086957  226747.562500  
2008    195445.833333  166690.636364  216042.375000  187036.363636  
2009    200051.947368  174463.120000  164525.000000  169837.000000  
2010              NaN            NaN            NaN            NaN

In [ ]:
#Jawaban nomor 2 menggunakan pivot_table dengan tahun menjadi kolom dan bulan menjadi baris
pd.pivot_table(
df,
index=["MoSold"],
columns=["YrSold"],
values="SalePrice",
aggfunc="mean")

YrSold           2006           2007           2008           2009  \
MoSold                                                               
1       201090.000000  195229.166667  196187.363636  213282.600000   
2       194550.000000  176301.750000  159370.000000  187450.000000   
3       190502.291667  179507.500000  183379.666667  181712.500000   
4       184247.750000  173844.090909  166244.652174  185028.000000   
5       162907.857143  173342.951220  193787.058824  171852.029412   
6       179218.181818  182400.403509  174276.625000  193688.627451   
7       189463.564516  200661.229167  186240.418605  203379.807018   
8       206477.095238  190515.083333  199285.071429  167971.444444   
9       206538.071429  205737.200000  195445.833333  200051.947368   
10      178139.136364  220190.733333  166690.636364  174463.120000   
11      213285.000000  197614.086957  216042.375000  164525.000000   
12      190140.000000  226747.562500  187036.363636  169837.000000   

YrSold           2010  
MoSold                 
1       167730.666667  
2       178989.285714  
3       203181.285714  
4       181439.909091  
5       186087.627907  
6       184072.142857  
7       135100.000000  
8                 NaN  
9                 NaN  
10                NaN  
11                NaN  
12                NaN

**Tabel pivot rata-rata SalePrice berdasarkan YrSold dan MoSold untuk tahun yang berbeda memberikan beberapa informasi, yakni sebagai berikut:**

1. **Analisis Rata - Rata SalePrice secara umum**

* Harga jual rata-rata secara umum turun dari tahun 2007 ke tahun 2010. Contohnya, harga rata-rata pada bulan Januari 2010 lebih rendah daripada harga pada bulan Januari 2007, yang menunjukkan bahwa nilai pasar secara umum telah turun secara signifikan.

2. **Analisis MoSold**

* Tabel menunjukkan bahwa harga jual berubah seiring bulan. Secara umum, harga jual rata-rata lebih tinggi dari bulan April hingga Juni (bulan ke-4 hingga ke-6) daripada bulan lainnya. Hal ini mengindikasikan bahwa bulan April hingga Bulan Juni adalah waktu yang bagus untuk menjual rumah. Hal ini mungkin dikarenakan ada lebih banyak pembeli pada waktu itu.

3. **Analisis YrSold**

* Dari tahun ke tahun, harga jual rata-rata berubah dengan jelas. Misalnya, harga pada bulan Januari dan Februari 2007 memiliki nilai yang jauh lebih tinggi daripada harga pada bulan yang sama di tahun 2008 dan 2009. Hal ini mengindikasikan adanya titik tertinggi pasar pada tahun 2007 dan penurunan harga pada tahun-tahun berikutnya. Hal ini mungkin dikarenakan kemerosotan ekonomi yang dimulai sekitar tahun 2008.

4. **Missing Values**

* Untuk beberapa bulan di tahun 2010, seperti Juli, Agustus, September, Oktober, November, dan Desember, tidak ada nilai. Hal Ini bisa berarti bahwa tidak ada cukup data penjualan atau bahwa ada lebih sedikit transaksi selama bulan-bulan tersebut, yang bisa berarti bahwa penelitian lebih lanjut perlu dilakukan.

## **Soal 3**

**Identifikasi Properti dengan Luas Tanah di Atas Persentil ke-90**

Lakukan Filtering Rumah dengan luas tanah (LotArea) yang lebih besar dari persentil ke-90. Setelah itu, urutkan properti tersebut berdasarkan harga penjualan (SalePrice) secara menurun. Tampilkan properti-properti dengan luas tanah yang paling besar, dan analisis apakah harga penjualan cenderung mengikuti ukuran tanah.




In [ ]:
#Mem-filter data dengan luas yanan yang lebih besar dari persentil ke-90
LotArea_bigger_than_90_persentil = df[(df['LotArea'] > df.LotArea.quantile(0.9))]
#Mengurutkan properti berdasarkan harga penjualan secara menurun
LotArea_bigger_than_90_persentil.sort_values(by='SalePrice', ascending=False)




Id  MSSubClass MSZoning  LotArea Street LotShape LandContour  \
632    692          60       RL    21535   Pave      IR1         Lvl   
1081  1183          60       RL    15623   Pave      IR1         Lvl   
1070  1170          60       RL    35760   Pave      IR1         Lvl   
955   1047          60       RL    16056   Pave      IR1         Lvl   
402    441          20       RL    15431   Pave      Reg         Lvl   
...    ...         ...      ...      ...    ...      ...         ...   
607    667          60       RL    18450   Pave      IR1         Lvl   
386    423          20       RL    21750   Pave      Reg         HLS   
1266  1384          30       RL    25339   Pave      Reg         Lvl   
599    659          50       RL    17503   Pave      Reg         Lvl   
968   1062          30  C (all)    18000   Grvl      Reg         Low   

     Utilities LotConfig LandSlope  ... EnclosedPorch 3SsnPorch ScreenPorch  \
632     AllPub    Corner       Gtl  ...             0         0           0   
1081    AllPub    Corner       Gtl  ...             0         0           0   
1070    AllPub   CulDSac       Gtl  ...             0         0           0   
955     AllPub    Inside       Gtl  ...             0         0           0   
402     AllPub    Inside       Gtl  ...             0         0         170   
...        ...       ...       ...  ...           ...       ...         ...   
607     AllPub    Inside       Gtl  ...             0         0           0   
386     AllPub    Inside       Mod  ...             0         0           0   
1266    AllPub    Inside       Gtl  ...           112         0           0   
599     AllPub    Inside       Gtl  ...             0         0           0   
968     AllPub    Inside       Gtl  ...             0         0           0   

     PoolArea MiscVal  MoSold  YrSold  SaleType  SaleCondition SalePrice  
632         0       0       1    2007        WD         Normal    755000  
1081      555       0       7    2007        WD        Abnorml    745000  
1070        0       0       7    2006        WD         Normal    625000  
955         0       0       7    2006       New        Partial    556581  
402         0       0       4    2009        WD         Normal    555000  
...       ...     ...     ...     ...       ...            ...       ...  
607         0       0       8    2007        WD        Abnorml    129000  
386         0       0       2    2008        WD         Normal    113000  
1266        0       0       8    2007        WD         Normal    112000  
599         0       0       1    2010        WD        Abnorml     97500  
968         0     560       8    2008     ConLD         Normal     81000  

[134 rows x 71 columns]

**Terlampir tabel yang mengidentifikasi properti dengan luas tanah di atas persentil ke-90 pada tabel di atas. Dari tabel tersebut didapatkan terdapat 134 entri data dengan luas tanah di atas persentil ke-90**

## **Soal 4**

**Analisis Zona Penjualan**

Anda ingin menganalisis jumlah total penjualan properti di setiap zona (MSZoning) berdasarkan tahun penjualan (YrSold). Gunakan tabel silang (crosstab) untuk menghitung jumlah penjualan di setiap zona untuk tiap tahun. Setelah itu, tambahkan kolom baru yang menandai zona dengan penjualan tinggi, di mana zona tersebut memiliki total penjualan di atas rata-rata.


In [ ]:
#Membuat crosstab untuk menghitung jumlah penjualan di setiap zona untuk tiap tahun
tabel_analisis_zona_penjualan = pd.crosstab(index=df["YrSold"], columns=df["MSZoning"], values=df["SalePrice"], aggfunc='sum')
#Menyimpan nilai rata - rata per tahun dalam variabel rata - rata per tahun
rata_rata_per_tahun = tabel_analisis_zona_penjualan.mean(axis = 1)

#Membuat fungsi untuk mengidentifikasi zona dengan penjualan tinggi (zona tersebut memiliki total penjualan di atas rata - rata)
def diatas_rata_rata(row):
  for zone, sales in row.items():
    if sales > rata_rata_per_tahun[row.name]:
      return zone

#Membuat kolom baru dengan nama kolom Zona Penjualan Tinggi (di atas rata-rata) dimana nilai entri data adalah hasil fungsi pengidentidaksi zona dengan penjualan tinggi terhadap data
tabel_analisis_zona_penjualan["Zona Penjualan Tinggi (di atas rata-rata)"] = tabel_analisis_zona_penjualan.apply(diatas_rata_rata, axis = 1)

#Menampilkan hasil tahap akhir
print(tabel_analisis_zona_penjualan)

MSZoning   C (all)         FV        RH          RL         RM  \
YrSold                                                           
2006      143311.0  2419302.0  226900.0  45732998.0  6023619.0   
2007      133900.0  3073952.0  423000.0  49782430.0  4923071.0   
2008      121000.0  2706730.0  321000.0  40429833.0  6816500.0   
2009           NaN  3468980.0  421034.0  48243083.0  5087079.0   
2010      227169.0  1654100.0       NaN  25259274.0  2249000.0   

MSZoning Zona Penjualan Tinggi (di atas rata-rata)  
YrSold                                              
2006                                            RL  
2007                                            RL  
2008                                            RL  
2009                                            RL  
2010                                            RL  


**Tabel pivot MSZoning dan SalePrice untuk tahun yang berbeda (Dari 2006 sampai 2010) memberikan beberapa informasi, yakni sebagai berikut:**

1. **Analisis Zona Penjualan Tinggi (di atas rata - rata)**

* Tabel menunjukkan rata-rata harga jual untuk zona C (semua), FV, RH, RL, dan RM dari tahun 2006 samapai 2010. Walaupun rata-rata harga jual bervariasi, tetapi properti pada Zona RL selalu memiliki harga tertinggi. Hal ini berarti properti dalam klasifikasi zona ini dijual dengan harga lebih tinggi daripada yang lain.

2. **Analisis YrSold**

* Sebagian besar jenis tanah pada tahun 2010 dijual dengan harga yang lebih rendah dari 2006 hingga 2009. Tren rata - rata harga jual kemudian cenderung stabil atau berfluktuasi pada 2010.

3. **Missing Values**:

* Zonasi C (semua) dan RH tidak memiliki nilai rata - rata pada 2009 (untuk Zonasi C) dan 2010 (untuk Zonasi RH). Hal ini mungkin dikarenakan tidak adanya kegiatan perdagangan pada tahun 2009 pada Zonasi C atau 2010 pada Zonasi RH. Maka dari itu, hal ini juga mengindikasikan bahwa Zonasi C dan Zonasi RH mungkin memiliki sedikit properti untuk dijual atau tidak mampu untuk menjual properti yang dimilikinya saat itu, sehingga mengindikasikan ketidakstabilan pasar atau perubahan minat pembeli.

## **Soal 5**

**Analisis Harga per Meter Persegi**

Anda diminta untuk melakukan feature engineering untuk membuat kolom baru bernama price_per_sqft, yang menghitung harga per meter persegi (dari SalePrice dibagi dengan LotArea). Setelah itu, urutkan hasilnya berdasarkan price_per_sqft secara menurun untuk menemukan properti dengan harga per meter persegi tertinggi. Analisis hasil tersebut dan identifikasi properti mana yang memiliki nilai tanah tertinggi.



In [ ]:
#Untuk menampilkan seluruh kolom
pd.set_option('display.max_columns', None)

In [ ]:
#Melakukan Feature Engineering dengan mendefinisikan kolom baru
df['price_per_sqft'] = df['SalePrice'] / df['LotArea']
#Mengurutkan data berdasarkan nilai price per sqft secara menurun
df.sort_values(by= "price_per_sqft", ascending=False)

Id  MSSubClass MSZoning  LotArea Street LotShape LandContour  \
879    957         160       RM     1300   Pave      Reg         Lvl   
1242  1359         160       FV     2117   Pave      Reg         Lvl   
162    181         160       FV     2117   Pave      Reg         Lvl   
570    624         160       FV     2117   Pave      Reg         Lvl   
693    759         160       FV     2280   Pave      Reg         Lvl   
...    ...         ...      ...      ...    ...      ...         ...   
646    707          20       RL   115149   Pave      IR2         Low   
1188  1299          60       RL    63887   Pave      IR3         Bnk   
285    314          20       RL   215245   Pave      IR3         Low   
226    250          50       RL   159000   Pave      IR2         Low   
306    336         190       RL   164660   Grvl      IR1         HLS   

     Utilities LotConfig LandSlope Neighborhood Condition1 Condition2  \
879     AllPub    Inside       Gtl      Blueste       Norm       Norm   
1242    AllPub    Inside       Gtl      Somerst       Norm       Norm   
162     AllPub    Inside       Gtl      Somerst       Norm       Norm   
570     AllPub    Inside       Gtl      Somerst       Norm       Norm   
693     AllPub    Inside       Gtl      Somerst       Norm       Norm   
...        ...       ...       ...          ...        ...        ...   
646     AllPub   CulDSac       Sev      ClearCr       Norm       Norm   
1188    AllPub    Corner       Gtl      Edwards      Feedr       Norm   
285     AllPub    Inside       Sev       Timber       Norm       Norm   
226     AllPub   CulDSac       Sev      ClearCr       Norm       Norm   
306     AllPub    Corner       Sev       Timber       Norm       Norm   

     BldgType HouseStyle  OverallQual  OverallCond  YearBuilt  YearRemodAdd  \
879    TwnhsE     2Story            6            6       1980          1980   
1242    Twnhs     2Story            6            5       2000          2000   
162     Twnhs     2Story            6            5       2000          2000   
570    TwnhsE     2Story            6            5       2000          2000   
693     Twnhs     2Story            7            5       1999          1999   
...       ...        ...          ...          ...        ...           ...   
646      1Fam     1Story            7            5       1971          2002   
1188     1Fam     2Story           10            5       2008          2008   
285      1Fam     1Story            7            5       1965          1965   
226      1Fam     1.5Fin            6            7       1958          2006   
306    2fmCon     1.5Fin            5            6       1965          1965   

     RoofStyle RoofMatl Exterior1st Exterior2nd  MasVnrArea ExterQual  \
879      Gable  CompShg     MetalSd     MetalSd         0.0        TA   
1242     Gable  CompShg     MetalSd     MetalSd       216.0        Gd   
162      Gable  CompShg     MetalSd     MetalSd       456.0        Gd   
570      Gable  CompShg     MetalSd     MetalSd       513.0        Gd   
693      Gable  CompShg     MetalSd     MetalSd       360.0        TA   
...        ...      ...         ...         ...         ...       ...   
646      Gable  CompShg     Plywood     Plywood       351.0        TA   
1188       Hip  ClyTile      Stucco      Stucco       796.0        Ex   
285        Hip  CompShg     BrkFace     BrkFace         0.0        TA   
226      Gable  CompShg     Wd Sdng     HdBoard       472.0        Gd   
306      Gable  CompShg     Plywood     Plywood         0.0        TA   

     ExterCond Foundation BsmtQual BsmtCond BsmtExposure BsmtFinType1  \
879         TA     CBlock       Gd       TA           No          ALQ   
1242        TA      PConc       Gd       TA           No          GLQ   
162         TA      PConc       Gd       TA           No          GLQ   
570         TA      PConc       Gd       TA           No          GLQ   
693         TA      PConc       Gd       TA           No          ALQ   
..

**Analisis**

**Properti dengan harga per meter persegi tertinggi adalah properti yang memiliki price per sqft 95.384615 dengan spesifikasi yang dapat dilihat di tabel dibawah sebagai berikut:**

In [ ]:
# Menggunakan iloc untuk menampilkan baris pertama data yang mengandung informasi properti dengan harga per meter persegi tertinggi.
df.sort_values(by= "price_per_sqft", ascending=False).iloc[[0]]

Id  MSSubClass MSZoning  LotArea Street LotShape LandContour Utilities  \
879  957         160       RM     1300   Pave      Reg         Lvl    AllPub   

    LotConfig LandSlope Neighborhood Condition1 Condition2 BldgType  \
879    Inside       Gtl      Blueste       Norm       Norm   TwnhsE   

    HouseStyle  OverallQual  OverallCond  YearBuilt  YearRemodAdd RoofStyle  \
879     2Story            6            6       1980          1980     Gable   

    RoofMatl Exterior1st Exterior2nd  MasVnrArea ExterQual ExterCond  \
879  CompShg     MetalSd     MetalSd         0.0        TA        TA   

    Foundation BsmtQual BsmtCond BsmtExposure BsmtFinType1  BsmtFinSF1  \
879     CBlock       Gd       TA           No          ALQ         285   

    BsmtFinType2  BsmtFinSF2  BsmtUnfSF  TotalBsmtSF Heating HeatingQC  \
879          Unf           0        276          561    GasA        TA   

    CentralAir Electrical  1stFlrSF  2ndFlrSF  LowQualFinSF  GrLivArea  \
879          Y      SBrkr       561       668             0       1229   

     BsmtFullBath  BsmtHalfBath  FullBath  HalfBath  BedroomAbvGr  \
879             0             0         1         1             2   

     KitchenAbvGr KitchenQual  TotRmsAbvGrd Functional  Fireplaces  \
879             1          TA             5        Typ           1   

    GarageFinish  GarageCars  GarageArea GarageQual PavedDrive  WoodDeckSF  \
879          Fin           2         462         TA          Y         150   

     OpenPorchSF  EnclosedPorch  3SsnPorch  ScreenPorch  PoolArea  MiscVal  \
879            0              0          0            0         0        0   

     MoSold  YrSold SaleType SaleCondition  SalePrice  price_per_sqft  
879       5    2009       WD        Normal     124000       95.384615

**Berikut tambahan analisis yang didapatkan dari price per sqft**

**1. Menyelidiki rata - rata price_per_sqft untuk setiap zonasi**

* Didapatkan Zonasi FV memiliki rata - rata price_per_sqft tertinggi dengan price_per_sqft 39.901495. Kemudian, diikuti Zonasi RM dengan rata - rata price_per_sqft 26.275912. Selanjutnya, Zonasi RH dengan price_per_sqft 20.816194. Setelah itu, zonasi RL 19.676881. Kemudian, terakhir Zonasi C (all) dengan price_per_sqft 8.606666.


**2. Menyelidiki hubungan rata - rata price_per_sqft untuk setiap zonasi terhadap waktu**
* Walaupun Zonasi RM bukan merupakan Zonasi dengan rata - rata price_per_sqft terbaik (Zonasi RM memiliki urutan kedua), namun dapat dilihat bahwa pada tahun 2009 (tahun dimana harga properti dengan harga per meter persegi tertinggi terjual) diketahui tahun itu adalah tahun dimana properti pada Zonasi RM memiliki penjualan tertinggi dari tahun 2006 sampai 2010

**3. Menyelidiki rata - rata price_per_sqft untuk setiap jenis jalan**

* Didapatkan properti yang memiliki jenis jalan pave memiliki rata - rata price_per_sqft tertinggi dengan price_per_sqft 21.556911. Sedangkan, properti yang memiliki jenis jalan Grvl memiliki rata - rata price_per_sqft yang rendah. Maka dari itu, hal ini mengindikasikan bahwa pembeli properti lebih menginginkan rumah dengan jenis jalan pave.

In [ ]:
#Menyelidiki rata - rata price_per_sqft untuk setiap zone dengan menggunakan groupby
df.groupby(["MSZoning"]).agg({
    "price_per_sqft": "mean"
}).sort_values(by = "price_per_sqft", ascending = False)

price_per_sqft
MSZoning                
FV             39.901495
RM             26.275912
RH             20.816194
RL             19.676881
C (all)         8.606666

In [ ]:
#Membuat tabel_analisis_price_per_sqft untuk setiap tahun dan zone dengan menggunakan crosstab
tabel_analisis_price_per_sqft = pd.crosstab(index=df["YrSold"], columns=df["MSZoning"], values=df["price_per_sqft"], aggfunc='mean')
tabel_analisis_price_per_sqft


MSZoning    C (all)         FV         RH         RL         RM
YrSold                                                         
2006       6.853027  37.662503  13.697133  19.385785  27.140511
2007      18.597222  43.155203  25.406756  19.785354  21.659560
2008       4.602941  41.955981  28.049587  19.153522  27.772675
2009            NaN  35.431327  17.316107  20.187492  30.371665
2010       9.114725  43.145705        NaN  19.866038  21.348079

In [ ]:
#Menyelidiki rata - rata price_per_sqft untuk setiap jenis jalan dengan fungsi groupby
df.groupby(["Street"]).agg({
    "price_per_sqft": "mean"
})

price_per_sqft
Street                
Grvl          6.123118
Pave         21.556911